### PR TITLE
Changing logic to return payload as json in the case of any exception depending on received bytes

### PR DESCRIPTION
**Pull request by nsarras**
tested on 4 node vm cluster successfully.

### CODE REVIEW

File: dss_datamover/socket_communication.py [hunk 1]
```diff
@@ -377,21 +373,19 @@ def recv_json(self, format="JSON"):
...
-            # return status depending on received message size, if incomplete a Runtime Error should have been raised
-            if len(msg_len_in_bytes) == response_header_length and received_data_size == msg_len:
-                return json.loads(msg)
-            else:
-                return json.loads("{}")
         except socket.error as e:
             self.logger.error(f"ServerSocket SocketError: {e}")
-            raise socket.error(f"ServerSocket SocketError: {e}")
         except ValueError as e:
-            raise ValueError(f"ServerSocket: ValueError - {e}")
```
Comment (grandsuri) [inline dss_datamover/socket_communication.py:389]: the top comment applies here as well

File: dss_datamover/socket_communication.py [hunk 2]
```diff
@@ -200,21 +199,19 @@ def recv_json(self, format="JSON"):
...
         except socket.timeout as e:
             self.logger.error("ClientSocket: Timeout ({} seconds) from recv function".format((datetime.now() - time_started).seconds))
-            # return status depending on received message size, if incomplete a Runtime Error should have been raised
-            if len(msg_len_in_bytes) == response_header_length and received_data_size == msg_len:
-                return json.loads(msg)
-            else:
-                return json.loads("{}")
         except socket.error as e:
             self.logger.error(f"ClientSocket SocketError: {e}")
-            raise socket.error(f"ClientSocket SocketError: {e}")
```
[the remark this answers is not in the document]
Comment (nsarras) [reply dss_datamover/socket_communication.py:210]: we could potentially do that, only downside is that we would lose how long a socket timeout took. If you think it is worth it to simplify the code, then I can go ahead and do that.

File: dss_datamover/socket_communication.py (hunk 1 above)
Comment (nsarras) [reply dss_datamover/socket_communication.py:389]: responded to comment above

File: dss_datamover/socket_communication.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (grandsuri) [reply dss_datamover/socket_communication.py:210]: Except socket.timeout, you can bundle rest of the exceptions to a generic exception unless it has some value.

[the remark this answers is not in the document]
Comment (nsarras) [reply dss_datamover/socket_communication.py:210]: sounds good, thank you

**Review by grandsuri — APPROVED** [review]

**Pull request merged**